### ISSUE TITLE
Handle query parameters correctly

### ISSUE THREAD

**Issue by thatside** [dotansimha/graphql-code-generator#555]
Schema: 
```graphql
type Query {
    search(query: String!): [FoundEntity!]!
}
```
Query:
```graphql
query Search($queryString: String!) {
  search(query: $queryString) {
    id # which is present in FoundEntity
  }
}
```

Produced output:
```graphql
export interface SearchQueryArgs {
  query: string | null;
}

export namespace Search {
  export type Variables = {
    queryString: string;
  };

  #... omitted
}
```

Expected output:
```graphql
export interface SearchQueryArgs {
  query: string;
}

export namespace Search {
  export type Variables = {
    queryString: string;
  };

  #... omitted
}
```

Seems to be incorrect behavior to me according to https://graphql.org/learn/schema/#arguments.

**Comment by thatside** [comment]
Also (may be related):
same schema and query;
Produced output:
```graphql
export namespace Search {
  #... omitted
  export type Query = {
    __typename?: "Query";
    search: (Search | null)[] | null;
  };
  #... omitted
}
```
Expected output:
```graphql
export namespace Search {
  #... omitted
  export type Query = {
    __typename?: "Query";
    search: Search[];
  };
  #... omitted
}
```

**Comment by dotansimha** [comment]
@thatside not sure I got it. the issue is that we don't treat `!` correctly?  thanks!

**Comment by thatside** [comment]
@dotansimha yes, exactly - both in case of query arguments and query return types.

**Comment by thatside** [comment]
Seems it was fixed after 12.0 - just cloned the repo to play with it and found that everything is ok :)
